c/src/ml-api-common.c

```diff
@@ -965,10 +1075,60 @@ ml_error (void)
  * @brief Internal interface to write messages for ml_error()
  */
 void
-_ml_error_report (const char *message)
+_ml_error_report_ (const char *fmt, ...)
+{
+  va_list arg_ptr;
+  G_LOCK (errlock);
+
+  va_start (arg_ptr, fmt);
+  vsnprintf (errormsg, _ML_ERRORMSG_LENGTH, fmt, arg_ptr);
+  va_end (arg_ptr);
+
+  errormsg[_ML_ERRORMSG_LENGTH - 1] = '\0';
```

Comment:
is it necessary?
(need to handle error code?)
```
int n;
n = vsnprintf (errormsg, _ML_ERRORMSG_LENGTH, fmt, arg_ptr);

n = (n > _ML_ERRORMSG_LENGTH || n < 0) ? _ML_ERRORMSG_LENGTH - 1 : n;
errormsg[n] = '\0';
```

Reply:
```errormsg[_ML_ERRORMSG_LENGTH - 1] = '\0';``` is actually not needed. vsnprintf will do it.
And let's not handle the truncate exception of exception handler :)
If it's truncated even with 4k buffer, let's keep it truncated.

Reply:
Code updated.